src/tron/TronEngine.ts

```diff
@@ -1490,9 +1427,13 @@ export class TronEngine extends CurrencyEngine<TronTools, SafeTronWalletInfo> {
 
   async getMaxSpendable(spendInfo: EdgeSpendInfo): Promise<string> {
     spendInfo = upgradeMemos(spendInfo, this.currencyInfo)
-    const { memos = [] } = spendInfo
+    const { memos = [], tokenId } = spendInfo
+    const currencyCode =
+      tokenId === null
+        ? this.currencyInfo.currencyCode
+        : this.allTokensMap[tokenId]
```

Comment:
I see this block in a bunch of places, and it seems like it will return `string | EdgeToken`? Did you mean `const {currencyCode} = tokenId == null ? this.currencyInfo : this.allTokensMap[tokenId]`, to actually return a `string` in all cases?